File: Java/Algorithms/Sorting/SelectionSort.java
public class SelectionSort {
    // Corrected swap method to swap elements in an array
    private void swap(int[] arr, int i, int j) { // Use array and indices for swap
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    // Selection Sort implementation
    public void sort(int[] arr) {
        int n = arr.length;
        
        for (int i = 0; i < n - 1; i++) { // Loop through array
            int minIndex = i; // Assume the current index is the minimum

            // Find the minimum element in the unsorted part
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j; // Update minIndex if a smaller element is found
                }
            }

            // Swap the minimum element with the current index
            swap(arr, i, minIndex); // Corrected swap method
        }
    }

    public static void main(String[] args) { // Corrected main method signature
        SelectionSort sorter = new SelectionSort(); // Create an instance of SelectionSort
        int[] arr = {3, 1, 6, 1, 8, 3, 4}; // Sample array

        sorter.sort(arr); // Call the sort method on the instance

        // Print the sorted array
        for (int i = 0; i < arr.length; i++) {
            System.out.println(arr[i]);
        }
    }
}

File: C++/Algorithms/Sorting/Selection Sort/selectionSort.cpp
#include <iostream>

using namespace std;

void selectionSort(int* arr, int size) {
    for (int i = 0; i < size - 1; i++) {
        int minIndex = i;  
        for (int j = i + 1; j < size; j++) {
            if (arr[j] < arr[minIndex]) {  
                minIndex = j;
            }
        }
        swap(arr[i], arr[minIndex]);
    }
}

int main(){
    int arr[] = {64, 25, 12, 22, 11};  // Sample array to sort
    int size = sizeof(arr) / sizeof(arr[0]);  // Calculate the size of the array

    selectionSort(arr, size);  
    for(int i=0; i<5; i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}